python/ray/train/base_trainer.py

```diff
@@ -182,6 +182,8 @@ def __init__(
 
         self._validate_attributes()
 
+        air_usage.tag_air_trainer(self)
```

Comment:
Should we move this somewhere later in the pipeline, e.g. `fit`? 

@jjyao what happens if you call this before auto initialization?

Reply:
I had some wondering. When I quickly tried it, it at least didn't crash even before `ray.init` was called.

Reply:
try selling my idea 1 more time, if we use the decorator approach, you can actually switch where these telemetries get sent without updating any Trainer code :)

Reply:
Yea we handle this case correctly: if ray.init() is not called yet, we will save tags to a local buffer and upload to gcs after ray.init() is called.